pkg/identity/numericidentity.go

```diff
@@ -23,9 +23,24 @@ const (
 	// shifted
 	ClusterIDShift = 16
 
-	// LocalIdentityFlag is the bit in the numeric identity that identifies
-	// a numeric identity to have local scope
-	LocalIdentityFlag = NumericIdentity(1 << 24)
+	// Identities also have scopes, which is defined by the high 8 bits.
+	// 0x00 -- global and reserved identities
+	// 0x01 -- local CIDR identities
+	// 0x02 -- remote nodes
+
+	// IdentityScopeMask is the top 8 bits of the 32 bit identity
+	IdentityScopeMask = NumericIdentity(0xFF_00_00_00)
+
+	// IdentityScopeGlobal is the identity scope used by global and reserved identities.
+	IdentityScopeGlobal = NumericIdentity(0)
+
+	// IdentityScopeLocalCIDR is the tag in the numeric identity that identifies
+	// a numeric identity to have local scope.
+	IdentityScopeLocalCIDR = NumericIdentity(1 << 24)
+
+	// IdentityScopeRemoteNode is the tag in the numeric identity that identifies
+	// an identity to be a remote in-cluster node.
+	IdentityScopeRemoteNode = NumericIdentity(2 << 24)
```

Comment:
Should we define min/max remote node scoped identities? Do we need the full 16 million?

Reply:
Related: does the allocator bound allocations within this range?

Reply:
Yeah, the allocated is bounded, so if we have 16 million node CIDRs allocation will fail.

From a resource perspective, we don't need the full 16 million. However, we do need a fast datapath "is-identity-remote-node" function. Right now that's just a single bitmask & compare.

I could shrink the space if you like and make it a simple range comparison, but I'm not super concerned about running out of ID scopes -- we have 254 more.

Reply:
I was idly wondering about bounding the section smaller (only allocate the first 65K or so) while retaining the mask, but then it would mismatch with the datapath. So yeah from that perspective maybe it's better to just leave this as-is for now, and if we run out in some hypothetical future then we can always come up with some migration techniques to reclaim bitspace.